src/pages/ReimbursementAccount/BankAccountStep.js

```diff
@@ -70,13 +88,42 @@ class BankAccountStep extends React.Component {
      */
     canSubmitManually() {
         return this.state.hasAcceptedTerms
+            && this.state.accountNumber.trim()
+            && this.state.routingNumber.trim();
+    }
 
-            // These are taken from BankCountry.js in Web-Secure
-            && CONST.BANK_ACCOUNT.REGEX.IBAN.test(this.state.accountNumber.trim())
-            && CONST.BANK_ACCOUNT.REGEX.SWIFT_BIC.test(this.state.routingNumber.trim());
+    /**
+     * @returns {Boolean}
+     */
+    validate() {
+        const errors = {};
+
+        // These are taken from BankCountry.js in Web-Secure
+        if (!CONST.BANK_ACCOUNT.REGEX.IBAN.test(this.state.accountNumber.trim())) {
+            errors.accountNumber = true;
+        }
+        if (!CONST.BANK_ACCOUNT.REGEX.SWIFT_BIC.test(this.state.routingNumber.trim())) {
+            errors.routingNumber = true;
+        }
+        setBankAccountFormValidationErrors(errors);
+        return _.size(errors) === 0;
+    }
+
+    validateAndSetTextValue(inputKey, value) {
+        const errors = this.getErrors();
+        if (errors[inputKey]) {
+            const newErrors = {...errors};
+            delete newErrors[inputKey];
+            setBankAccountFormValidationErrors(newErrors);
+        }
```

Comment:
NAB, it might be good to leave a comment here that these lines exist to clear the existing error when a particular field changes. 

Also it might be a good opportunity for an early return if we move it after the `setState()` call (this is a really common pattern throughout our codebases).

```js
this.setState();
const errors = this.getErrors();
if (!errors[inputKey]) {
    return;
}

// Clear the error

Reply:
I like early returns, but I tend to use them when there is more nesting happening. I changed it anyway because I find it ok both ways.